src/views/TypingGame.vue

```diff
@@ -0,0 +1,192 @@
+<!-- src/views/TypingGame.vue -->
+<template>
+    <div class="typing-game">
+      <h1>KeyClash Typing Game</h1>
+      <div class="game-area">
+        <p class="text-to-type">{{ currentText }}</p>
+        <input
+          v-model="userInput"
+          @input="checkInput"
+          @keydown="handleKeydown"
+          :disabled="!gameStarted || gameEnded"
+          placeholder="Type here..."
+          :style="inputTextStyle"
+        />
+        <p v-if="gameEnded">Game Over! Your WPM: {{ wpm }}</p>
+        <button @click="startGame" v-if="!gameStarted" :class="{'play-btn': true}">Start Game</button>
+        <p v-if="pasteToWin">We don't allow PASTE TO WIN...</p>
+      </div>
+    </div>
+  </template>
+  
+  <script>
+  import { ref, computed } from 'vue';
+  
+  export default {
+    name: 'TypingGame',
+    setup() {
+      const texts = [
+        "The quick brown fox jumps over the lazy dog.",
+        "Programming is the art of telling another human what one wants the computer to do.",
+        "The only way to learn a new programming language is by writing programs in it.",
+      ];
+  
+      const currentText = ref('');
+      const userInput = ref('');
+      const gameStarted = ref(false);
+      const gameEnded = ref(false);
+      const pasteToWin = ref(false);
+      const pasted= ref(false);
+      const startTime = ref(0);
+      const endTime = ref(0);
+   
+      let curCharTextColor = "black";
+
+      const wpm = computed(() => {
+        if (!gameEnded.value) return 0;
+        const timeInMinutes = (endTime.value - startTime.value) / 60000;
+        const wordsTyped = currentText.value.split(' ').length;
+        return Math.round(wordsTyped / timeInMinutes);
+      });
+
+      function startGame() {
+        currentText.value = texts[Math.floor(Math.random() * texts.length)];
+        userInput.value = '';
+        gameStarted.value = true;
+        gameEnded.value = false;
+        startTime.value = Date.now();
+      }
+      let keysPresseedIterator = 0;
+      let backspacePressed = false;
+
+      function checkInput() {  
+        let userInputChar = userInput.value.split('');
+        let charsToType = currentText.value.split('') 
+
+          if (userInputChar[keysPresseedIterator] != undefined && (userInputChar[keysPresseedIterator] === charsToType[keysPresseedIterator])) {
```

Comment:
Will `charsToType[keysPresseedIterator]` ever be undefined? 

I think something like this is more readable.


```suggestion
      function checkInput() {  
        let userInputChar = userInput.value.split('');
        let charsToType = currentText.value.split('') 
        let inputChar = userInputChar[keysPresseedIterator]
        
        if (inputChar === charsToType[keysPresseedIterator])) {
```

Reply:
I added the undefined check because I was finding userInputChar[keysPresseedIterator] to be undefined. At one point, both pieces of data I was checking were equal because they were undefined. Might not be necessary at this point and I like the readability you mention here.